contracts/base/ModuleManager.sol

```diff
@@ -86,10 +105,22 @@ abstract contract ModuleManager is SelfAuthorized, Executor {
     ) public virtual returns (bool success) {
         // Only whitelisted modules are allowed.
         require(msg.sender != SENTINEL_MODULES && modules[msg.sender] != address(0), "GS104");
+
+        address moduleGuard = getModuleGuard();
+
+        if (moduleGuard != address(0)) {
+            ModuleGuard(moduleGuard).checkTransaction(to, value, data, operation, msg.sender);
+        }
+
         // Execute transaction without further confirmations.
         success = execute(to, value, data, operation, type(uint256).max);
         if (success) emit ExecutionFromModuleSuccess(msg.sender);
         else emit ExecutionFromModuleFailure(msg.sender);
+
+        if (moduleGuard != address(0)) {
+            bytes32 dataHash = keccak256(data);
```

Comment:
Not sure if the hash of data adds any value to the check. @rmeissner wdyt?

Reply:
agree